File: server/models/Batch.js
module.exports = (sequelize, DataTypes) => {
  const Batch = sequelize.define(
    'Batch',
    {
      batchId: {
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
        primaryKey: true,
      },
      batchName: {
        type: DataTypes.STRING,
        allowNull: false,
      },
      userId: {  // ✅ Maps to `user_id` in the database
        type: DataTypes.UUID,
        allowNull: false,
        references: {
          model: 'Users',
          key: 'userId',
        },
        field: 'user_id', // ✅ Ensures Sequelize maps `userId` to `user_id`
      },
      groups: {
        type: DataTypes.JSON,
        allowNull: false,
        defaultValue: [],
      },
      groupCount: {
        type: DataTypes.INTEGER,
        defaultValue: 0,
      },
      instructors: {
        type: DataTypes.JSON,
        allowNull: true,
        defaultValue: [],
      },
      instructorNames: {
        type: DataTypes.JSON,
        allowNull: true,
        defaultValue: [],
      },
      instructorIds: {
        type: DataTypes.JSON,
        allowNull: true,
        defaultValue: [],
      },
      createdAt: {
        type: DataTypes.DATE,
        allowNull: false,
        defaultValue: DataTypes.NOW,
      },
      visibility: {
        type: DataTypes.BOOLEAN,
        allowNull: false,
        defaultValue: true,
      },
    },
    {
      timestamps: true,
    }
  );

  Batch.associate = (models) => {
    Batch.belongsTo(models.User, {
      foreignKey: 'userId', // ✅ Use `userId` here
      onDelete: 'CASCADE',
    });
  };

  return Batch;
};

File: eps-server-go/models/batch.go
package models

import (
	"time"
	"github.com/google/uuid"
	"gorm.io/gorm"
	"gorm.io/datatypes"
)

type Batch struct {
	BatchID         uuid.UUID      `gorm:"type:uuid;primaryKey" json:"batchId"`
	BatchName       string         `gorm:"not null" json:"batchName"`
	UserID          uuid.UUID      `gorm:"type:uuid;column:user_id" json:"userId"` 
	Groups          datatypes.JSON `json:"groups"`          
	GroupCount      int            `gorm:"default:0" json:"groupCount"`
	Instructors     datatypes.JSON `json:"instructors"`     
	InstructorNames datatypes.JSON `json:"instructorNames"` 
	InstructorIds   datatypes.JSON `json:"instructorIds"`   
	Visibility      bool           `gorm:"default:true" json:"visibility"`
	CreatedAt       time.Time      `json:"createdAt"`
	UpdatedAt       time.Time      `json:"updatedAt"`

	User User `gorm:"foreignKey:UserID;constraint:OnDelete:CASCADE" json:"user,omitempty"`
}

func (b *Batch) BeforeCreate(tx *gorm.DB) (err error) {
	b.BatchID = uuid.New()
	return
}